zerver/views/development/email_log.py

```diff
@@ -112,10 +112,7 @@ def generate_all_emails(request: HttpRequest) -> HttpResponse:
     url = confirmation_url(key, realm, Confirmation.EMAIL_CHANGE)
     user_profile = get_user_by_delivery_email(registered_email, realm)
     result = client.get(url)
-    assert result.status_code == 200
-
-    # Reset the email value so we can run this again
-    do_change_user_delivery_email(user_profile, registered_email)
+    assert result.status_code == 302
```

Comment:
what's the story behind this change?

Reply:
We decided to redirect user to login page with a GET param but this is expecting a status code of 200 and with this change in email change flow, the email won't change just on sending a request to that link. So, removed the `do_change_user_delivery_email` too.
Mentioned this in commit message too
```
In zerver/views/development/email_log.py, changed the
assertion of status to 302 as we redirect to login page
with this commit on clicking email change confirmation link.
Also removed the change of delivery email as it's no longer
needed since the email won't be changed on clicking the link.
```

Reply:
Changed it to authenticate using email+password.

Reply:
Have you tested manually whether this works? It looks right, but worth checking.

Reply:
yes, tested manually before pushing. it works :)